arangod/Cluster/SynchronizeShard.cpp

```diff
@@ -862,15 +866,39 @@ bool SynchronizeShard::first() {
     }
 
     auto ep = clusterInfo.getServerEndpoint(leader);
+    uint64_t docCountOnLeader = 0;
+    if (!collectionCountOnLeader(ep, docCountOnLeader).ok()) {
+      std::stringstream error;
+      error << "failed to get a count on leader " << database << "/" << shard;
+      LOG_TOPIC("1254a", ERR, Logger::MAINTENANCE) << "SynchronizeShard " << error.str();
+      result(TRI_ERROR_INTERNAL, error.str());
+      return false;
+    }
+
     uint64_t docCount = 0;
     if (!collectionCount(*collection, docCount).ok()) {
       std::stringstream error;
-      error << "failed to get a count on leader " << database << "/" << shard;
+      error << "failed to get a count here " << database << "/" << shard;
```

Comment:
I suggest to not dismiss the original error here, so that we have a better error message should this happen, and can also return the proper error number.

Reply:
Done.